server/connector/data_sink.cpp

```diff
@@ -120,16 +120,6 @@ void RocksDBDataSink::appendData(velox::RowVectorPtr input) {
   const auto num_columns = input->childrenSize();
   [[maybe_unused]] const auto& input_type = input->type()->asRow();
   for (velox::column_index_t i = 0; i < num_columns; ++i) {
-    // Exclude only UNKNOWN from equivalency check as UNKNOWN is just a NULL and
-    // compatible with everything.
-    // Intentionally check UNKNOWN after equivalency to make compiler execute
-    // child lookup by name as it is also a necessary check.
-    // Even keys during UPDATE should be validated as it is important for
-    // RocksDB key generation
-    VELOX_DCHECK(_row_type->findChild(input_type.nameOf(i))
-                   ->equivalent(*input_type.childAt(i)) ||
-                 input_type.childAt(i)->kind() == velox::TypeKind::UNKNOWN);
```

Comment:
this doesn't work, and quite fragile, so I removed it. should be checked in planning, not on execution